src/enemy.cairo

```diff
@@ -1,41 +1,206 @@
-use cubit::types::vec2::{Vec2, Vec2Trait};
-use cubit::types::fixed::{Fixed, FixedTrait};
-
-#[derive(Component, Serde, SerdeLen, Drop)]
-struct Enemy {
-    typ: u8, 
-}
+/// Number of enemies to spawn.
+const ENEMIES_NB: u8 = 10;
+/// Height of the grid.
+const GRID_HEIGHT: u128 = 1000;
 
 // Road dimensions
 // 400x1000
 
 #[system]
 mod spawn_enemies {
+    use traits::Into;
+    use array::{Array, ArrayTrait};
+    use cubit::types::vec2::Vec2;
+    use dojo::world::Context;
+    use drive_ai::Vehicle;
+    use super::ENEMIES_NB;
+
+    /// Spawn the enemies provided in the array.
+    /// /!\ Panics if the number of enemies provided in the array don't match the expected number
+    /// of enemies defined in the [`suer::ENEMIES_NB`] constant.
+    /// There's no sanity check on any of the properties of the enemies so it's possible to spawn
+    /// enemies with really high speed, on top of each other or too big for the grid.
+    ///
+    /// # Arguments
+    ///
+    /// * `ctx` - Context of the game.
+    /// * `enemies` - Array of enemies. It is assumed that all the enemies in this array are valid.
+    fn execute(ctx: Context, enemies: Array<Vehicle>) {
```

Comment:
we will need to pass in the `model` here, which name spaces a simulation instance